### ISSUE TITLE
[Airbrake] [Production] key not found: "buffer_space_heating_16"

### ISSUE THREAD

**Issue by dennisschoenmakers** [quintel/etmoses#1120]
**Airbrake error:** [#2024](https://quintel.airbrake.io/projects/111227/groups/1713010473251092024)
**Airbrake project:** ETMoses

**Error type:** `KeyError`
**Error message:** `` key not found: "buffer_space_heating_16" ``
**Where:** `testing_grounds#data`
**Occurred at:** Jun 21, 2016 12:04:16 UTC
**First seen at:** Jun 21, 2016 12:04:16 UTC
**Occurrences:** 1 (1 since last deploy on Jun 20, 2016 12:46:23 UTC)

**URL:** [https://moses.energytransitionmodel.com/testing_grounds/478/data.json](https://moses.energytransitionmodel.com/testing_grounds/478/data.json)
**File:** `[PROJECT_ROOT]/app/models/calculation/technology_load.rb`

**Backtrace:**
```
[PROJECT_ROOT]/app/models/calculation/technology_load.rb:88:in fetch
[PROJECT_ROOT]/app/models/calculation/technology_load.rb:88:in composite
[PROJECT_ROOT]/app/models/calculation/technology_load.rb:48:in block (2 levels) in techs_for
[PROJECT_ROOT]/app/models/installed_technology/profile_curve.rb:17:in block in each
```

**Comment by antw** [comment]
Just to be clear about this one...

This is not an error in the calculator. A heating technology is reporting that it belongs to `buffer_space_heating_16`, and no such buffer exists. I suspect something in the technology form is to blame.

```js
    {
      "buffer": "buffer_space_heating_16",
      "capacity": 10.0,
      "composite": false,
      "composite_value": null,
      "congestion_reserve_percentage": null,
      "demand": null,
      "name": "Heat pump for space heating (air)",
      "position_relative_to_buffer": "buffering",
      "profile": null,
      "type": "households_space_heater_heatpump_air_water_electricity",
      "units": 18,
      "volume": null,
      "composite_index": 2,
      "concurrency": "min",
      "full_load_hours": 1167,
      "includes": [
        
      ],
      "initial_investment": 9500.0,
      "om_costs_for_ccs_per_full_load_hour": null,
      "om_costs_per_full_load_hour": 0.0,
      "om_costs_per_year": 100.0,
      "performance_coefficient": 4.5,
      "technical_lifetime": 15,
      "associates": [
        
      ],
      "node": null,
      "profile_key": null
    },
```

**Comment by grdw** [comment]
> I suspect something in the technology form is to blame.

I think so to... my best guess would be that there's something going haywire when hitting the concurrency buttons too many times. However, it could also be a bit harder like adding 20 buffers and than adding one again, than switching concurrency? I need to look into this though.

**Comment by antw** [comment]
From @ChaelKruip via e-mail:

> I triggered this error by moving a technology to another buffer and deleting the original buffer it was attached to...
> Strangely, it seems not to be assigned to any buffer now:
>
> ![etmoses](https://cloud.githubusercontent.com/assets/4383/16560189/b864b09c-41e9-11e6-9473-51958f65d20d.png)

I can reproduce this. Apparently, if I move a technology from one buffer to another, save and then refresh, the technology is not actually moved; it remains attached to the original buffer.

**Comment by grdw** [comment]
> Apparently, if I move a technology from one buffer to another, save and then refresh, the technology is not actually moved; it remains attached to the original buffer.

👍 Thanks for finding out the problem though. I'll take a look pretty soon.

**Comment by dennisschoenmakers** [comment]
*Comment from Airbrake:* Resolved [error](https://quintel.airbrake.io/projects/111227/groups/1713010473251092024) reoccurred.

**Comment by grdw** [comment]
This has to do with the concurrency problems... this is obviously solved with #1277
